scripts_python/ae_train_model.py

```diff
@@ -0,0 +1,148 @@
+# Importing now to solve issues later.
+import comet_ml
+import torch
+
+from scilpy.io.utils import assert_inputs_exist, assert_outputs_exist
+
+from dwi_ml.data.dataset.utils import prepare_multisubjectdataset
+from dwi_ml.experiment_utils.prints import format_dict_to_str
+from dwi_ml.experiment_utils.timer import Timer
+from dwi_ml.io_utils import add_logging_arg, add_memory_args
+from dwi_ml.models.projects.ae_models import ModelAE
+from dwi_ml.training.trainers import DWIMLAbstractTrainer
+from dwi_ml.training.utils.batch_samplers import (add_args_batch_sampler,
+                                                  prepare_batch_sampler)
+from dwi_ml.training.utils.batch_loaders import (add_args_batch_loader)
+from dwi_ml.training.batch_loaders import DWIMLAbstractBatchLoader
+from dwi_ml.training.utils.experiment import (
+    add_mandatory_args_experiment_and_hdf5_path)
+from dwi_ml.training.utils.trainer import run_experiment, add_training_args, \
+    format_lr
+
+
+def prepare_arg_parser():
+    p = argparse.ArgumentParser(description=__doc__,
+                                formatter_class=argparse.RawTextHelpFormatter)
+    add_mandatory_args_experiment_and_hdf5_path(p)
+    add_args_batch_sampler(p)
+    add_args_batch_loader(p)
+    training_group = add_training_args(p)
+    p.add_argument('streamline_group_name',
+                   help="Name of the group in hdf5")
+    add_memory_args(p, add_lazy_options=True, add_rng=True)
+    add_logging_arg(p)
+
+    # Additional arg for projects
+    training_group.add_argument(
+        '--clip_grad', type=float, default=None,
+        help="Value to which the gradient norms to avoid exploding gradients."
+             "\nDefault = None (not clipping).")
+
+    return p
+
+
+def init_from_args(args, sub_loggers_level):
+    torch.manual_seed(args.rng)  # Set torch seed
+
+    # Prepare the dataset
+    dataset = prepare_multisubjectdataset(args, load_testing=False,
+                                          log_level=sub_loggers_level)
+
+    # Preparing the model
+    # (Direction getter)
+    # (Nb features)
+    # Final model
+    with Timer("\n\nPreparing model", newline=True, color='yellow'):
+        # INPUTS: verifying args
+        model = ModelAE(
+            experiment_name=args.experiment_name,
+            step_size=args.step_size, compress_lines=args.compress,
+            kernel_size=3, latent_space_dims=32,
+            log_level=sub_loggers_level)
+
+        logging.info("AEmodel final parameters:" +
+                     format_dict_to_str(model.params_for_checkpoint))
+
+        logging.info("Computed parameters:" +
+                     format_dict_to_str(model.computed_params_for_display))
+
+    # Preparing the batch samplers
+    batch_sampler = prepare_batch_sampler(dataset, args, sub_loggers_level)
+    # Preparing the batch loader.
+    with Timer("\nPreparing batch loader...", newline=True, color='pink'):
+        batch_loader = DWIMLAbstractBatchLoader(
```

Comment:
Si tu préfères le renommer DWIMLStreamlinesBatchLoader, pour être moins laid que abstract, go for it.